skcosmo/feature_selection/_greedy.py

```diff
@@ -0,0 +1,154 @@
+        next feature to select is that which maximizes the score.
+
+        NOTE that when using custom scorers, each scorer should return a single
+        value. Metric functions returning a list/array of values can be wrapped
+        into multiple scorers that return one value each.
+
+    Attributes
+    ----------
+    n_features_to_select_ : int
+        The number of features that were selected.
+
+    support_ : ndarray of shape (n_features,), dtype=bool
+        The mask of selected features.
+
+    """
+
+    def __init__(
+        self,
+        scoring,
+        n_features_to_select=None,
+    ):
+
+        self.n_features_to_select = n_features_to_select
+        self.scoring = scoring
+
+    def fit(self, X, y=None, current_mask=None):
+        """Learn the features to select.
+
+        Parameters
+        ----------
+        X : array-like of shape (n_samples, n_features)
+            Training vectors.
+        y : array-like of shape (n_samples,)
+            Target values.
+        current_mask : array-like of shape (n_features)
+                       an initial mask to start with, if restarting
+
+        Returns
+        -------
+        self : object
+        """
+        tags = self._get_tags()
+
+        if y is not None:
+            X, y = self._validate_data(
+                X,
+                y,
+                accept_sparse="csc",
+                ensure_min_features=2,
+                force_all_finite=not tags.get("allow_nan", True),
+                multi_output=True,
+            )
+        else:
+            X = check_array(
+                X,
+                accept_sparse="csc",
+                ensure_min_features=2,
+                force_all_finite=not tags.get("allow_nan", True),
+            )
+
+        n_features = X.shape[1]
+
+        error_msg = (
+            "n_features_to_select must be either None, an "
+            "integer in [1, n_features - 1] "
+            "representing the absolute "
+            "number of features, or a float in (0, 1] "
+            "representing a percentage of features to "
+            f"select. Got {self.n_features_to_select}"
+        )
+        if self.n_features_to_select is None:
+            self.n_features_to_select_ = n_features // 2
+        elif isinstance(self.n_features_to_select, numbers.Integral):
+            if not 0 < self.n_features_to_select < n_features:
+                raise ValueError(error_msg)
+            self.n_features_to_select_ = self.n_features_to_select
+        elif isinstance(self.n_features_to_select, numbers.Real):
+            if not 0 < self.n_features_to_select <= 1:
+                raise ValueError(error_msg)
+            self.n_features_to_select_ = int(n_features * self.n_features_to_select)
+        else:
+            raise ValueError(error_msg)
+
+        # the current mask corresponds to the set of features:
+        if current_mask is None:
+            current_mask = np.zeros(shape=n_features, dtype=bool)
```

Comment:
It's not so important, but maybe it's better to make mask boolean array? As I understood, it's boolean in `SelectorMixin`